apps/docs/pages/guides/auth/auth-postgres-hooks.mdx

```diff
@@ -0,0 +1,851 @@
+  src="/docs/img/guides/auth-postgres-hooks/hooks-flowchart.png"/>
+
+
+## Create a Hook
+### Write a Hook
+
+In the dashboard, go to `SQL Editor > Templates` and select `Add Auth Hook (General)`. The function for your hook should look like this:
+
+```sql
+create or replace function
+mfa_verification_attempt(event jsonb) returns jsonb as $$
+declare
+  -- Add your variables here
+begin
+  -- Add your logic here
+  return response
+end;
+$$ language plpgsql;
+```
+
+Note that the JSONB operator `?` can't be used when writing a hook, because it is a reserved character in the ORM used by Supabase Auth.
+
+### Handle Hook errors
+
+Unless otherwise specified, your hook should return errors with the following fields:
+
+- `error` An object that contains information about the error.
+    - `http_code` A number indicating the HTTP code to be returned. If not set, the code is HTTP 500 Internal Server Error.
+    - `message` A message to be returned in the HTTP response. Required.
+
+You should return an error if an input in the `event` payload has been processed incorrectly. This error could result from a server error (for example, an unexpected error with the data) or a user error (for example, the user has exceeded the number of available attempts).
+
+In general, Supabase Auth ignores all fields other than the `error` field attached to the returned JSON.
+
+Example output:
+```jsx
+{
+  "error": {
+    "http_code": 429,
+    "message": "You can only verify a factor once every 10 seconds."
+  }
+}
+```
+
+### Assign Hook permissions
+
+You need to assign additional permissions to the hook and the tables that it interacts with.
+
+By default, Supabase Auth invokes hooks with the `supabase_auth_admin` role. This role doesn't have permissions to modify the `public` schema. You need to grant the role permission to execute your hook function:
+
+```sql
+grant execute
+  on function public.custom_access_token_hook
+  to supabase_auth_admin;
+```
+
+Also revoke permissions from the `authenticated` and `anon` roles to ensure the function is not accessible by Supabase Serverless APIs.
+
+```sql
+revoke execute
+  on function public.custom_access_token_hook
+  from authenticated, anon;
+```
+
+<Admonition type="caution">
+Avoid adding a `security definer` tag to the function to avoid permission errors.
+</Admonition>
+
+At time of writing, Postgres functions have an implicit tag of `security invoker` and run with the permissions of the calling user. In the case of Auth Hooks, `supabase_auth_admin`  is the invoker and hook functions run with the permissions of `supabase_auth_admin`.
+
+The `security definer` tag executes a function with the privileges of the user who owns it.  In the case of functions created via the Supabase Dashboard, this user is `postgres`, which does not have permissions for the Auth schema. Using `security definer` tag would prevent Supabase Auth from having the appropriate permissions to access tables or execute functions.
```

Comment:
Maybe just link to Postgres docs. `security definer` with the correct grants can be easier to work with.

Reply:
Perhaps we can just link to https://supabase.com/docs/guides/database/functions#security-definer-vs-invoker ?

I think we already do a decent job of explaining this there